### PR TITLE
[202012]Dell s6100 Fix sonic-mgmt platform test failures

### PR DESCRIPTION
**Pull request by aravindmani-1**
ADO: 24709703

#### Why I did it
sonic-mgmt platform testcases failed.
#### How I did it
Implement platform API 2.0.
#### How to verify it

Run sonic-mgmt tests.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

### CODE REVIEW

**Comment by prgeor** [comment]
@aravindmani-1 Thanks. Do you mind attaching the sonic-mgmt test run to this PR?

File: platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/module.py [hunk 1]
```diff
@@ -198,3 +198,49 @@ def get_system_eeprom_info(self):
                   ‘0x26’:’01’, ‘0x27’:’REV01’, ‘0x28’:’AG9064-C2358-16G’}
         """
         return self._eeprom.system_eeprom_info()
+
```
Comment (prgeor) [inline platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/module.py:201]: @aravindmani-1 can you run the sonic-mgmt test without these changes? These test are supposed to run only for T2 chasssis.Ref:-https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml#L317

**Comment by aravindmani-1** [comment]
> @aravindmani-1 Thanks. Do you mind attaching the sonic-mgmt test run to this PR?

[Dell_S6100_platform_api_test_suite_report.xlsx](https://github.com/sonic-net/sonic-buildimage/files/12381627/Dell_S6100_platform_api_test_suite_report.xlsx)

File: platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/module.py (hunk 1 above)
Comment (aravindmani-1) [reply platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/module.py:201]: Without these changes, platform cases in module.py fails.

**Comment by prgeor** [comment]
> > @aravindmani-1 Thanks. Do you mind attaching the sonic-mgmt test run to this PR?
> 
> [Dell_S6100_platform_api_test_suite_report.xlsx](https://github.com/sonic-net/sonic-buildimage/files/12381627/Dell_S6100_platform_api_test_suite_report.xlsx)

@aravindmani-1 could you please run other platform tests (For eg test_sfputil.py )?

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@qiluo-msft please merge to 202012

**Comment by qiluo-msft** [comment]
@aravindmani-1 Is it on master already? Could you change PR description to mention the master PR link?

**Comment by aravindmani-1** [comment]
> @aravindmani-1 Is it on master already? Could you change PR description to mention the master PR link?
Master branch already has these changes. Multiple PR's are part of master branch API 2.0 implementation. Hence defined the PR description commonly.

**Pull request merged**